packages/runtime/src/templating/resources/with.ts

```diff
@@ -12,7 +12,7 @@ export interface With extends ICustomAttribute {}
 export class With {
   public static register: IRegistry['register'];
 
-  @bindable public value: any = null;
+  @bindable public value: IBindScope | IBindingContext = null;
```

Comment:
The only assumption we can make about this type is that it will be an object. For example:
```ts
export class MyViewModel {
  public person = { name: 'Bob' }
}
```
```html
<div with.bind="person">
  ${name}
</div>
```

So this might be one of the very rare situations where `IIndexable` is actually the most appropriate type.
That said, `IIndexable` introduces `any`. And changing that to `unknown` is something which I believe we can't do quite yet. Perhaps introduce an `IUnknownIndexable` as a temporary intermediate? Thoughts?

Reply:
I chose those types, because that's what `Scope.fromParent` on line 47 expects it to be. And as that's the only place the class that `value` is used... Does that mean we need to change `fromParent` as well?

Reply:
Oh. You're actually raising a really important point now. I never realized this particular scenario when I rewrote `Scope`, `OverrideContext` and `BindingContext`. I need to look into that.
Goes to show how important strict typing is as this might expose potential issues that otherwise might not have surfaced until it bit an end user. Good stuff :)

Reply:
So leave it as `IBindScope | IBindingContext` for now? Or go the `IUnknownIndexable` way? My preference would be option 1, because that is a valid representation of how it's currently implemented and then "fix" it when you look into it a bit more.

Reply:
```suggestion
  // TODO: this type is incorrect (it can be any user-provided object), need to fix and double check Scope
  @bindable public value: IBindScope | IBindingContext = null;
```
Yeah that's okay. I suggested a TODO comment so i don't forget